common/djangoapps/third_party_auth/api/views.py

```diff
@@ -89,3 +98,122 @@ def get(self, request, username):
+
+          GET /api/third_party_auth/v0/providers/{provider_id}/users?remote_ids={remote_id1},{remote_id2}
+
+          GET /api/third_party_auth/v0/providers/{provider_id}/users?usernames={username1}&remote_ids={remote_id1}
+
+    **URL Parameters**
+
+        * provider_id: The unique identifier of third_party_auth provider (e.g. "saml-ubc", "oa2-google", etc.
+          This is not the same thing as the backend_name.). (Optional/future: We may also want to allow
+          this to be an 'external domain' like 'ssl:MIT' so that this API can also search the legacy
+          ExternalAuthMap table used by Standford/MIT)
+
+    **Query Parameters**
+
+        * remote_ids: Optional. List of comma separated remote (third party) user IDs to filter the result set.
+          e.g. ?remote_ids=8721384623
+
+        * usernames: Optional. List of comma separated edX usernames to filter the result set.
+          e.g. ?usernames=bob123,jane456
+
+        * page, page_size: Optional. Used for paging the result set, especially when getting
+          an unfiltered list.
+
+    **Response Values**
+
+        If the request for information about the user is successful, an HTTP 200 "OK" response
+        is returned.
+
+        The HTTP 200 response has the following values:
+
+        * count: The number of mappings for the backend.
+
+        * next: The URI to the next page of the mappings.
+
+        * previous: The URI to the previous page of the mappings.
+
+        * num_pages: The number of pages listing the mappings.
+
+        * results:  A list of mappings returned. Each collection in the list
+          contains these fields.
+
+            * username: The edx username
+
+            * remote_id: The Id from third party auth provider
+    """
+    authentication_classes = (
+        OAuth2Authentication,
+    )
+
+    paginate_by = 50
+    paginate_by_param = 'page_size'
+    pagination_serializer_class = PaginationSerializer
+    serializer_class = serializers.UserMappingSerializer
+    provider = None
+
+    def get_queryset(self):
+        self.provider = Registry.get(self.kwargs.get('provider_id'))
+        if not self.provider:
+            raise Http404
+
+        # permission checking. We allow both API_KEY access and OAuth2 client credential access
+        if not (
+                ApiKeyHeaderPermission().has_permission(self.request, self) or
+                ThirdPartyAuthApiPermission().has_permission(self.request, self)
+        ):
+            raise exceptions.PermissionDenied()
+
+        usernames_str = self.request.QUERY_PARAMS.get('usernames', None)
+        remote_ids_str = self.request.QUERY_PARAMS.get('remote_ids', None)
+
+        query_set = UserSocialAuth.objects.select_related('user').filter(provider=self.provider.backend_name)
+
+        # build our query filters
+        # When using multi-IdP backend, we only retrieve the ones that are for current IdP
+        uid = self.provider.get_social_auth_uid('uid')
```

Comment:
This is basically testing if the there is any idp_slug for this provider. There are two types of provider, one like google, only one provider, one backend. Another is SAML, one SAML backend but multiple providers, e.g. UBC SAML, MIT SAML. Because social auth table doesn't have a column to support multi-provider backend. The implementation for SAML is prepending the slug in the remote_id column to distinguish between different providers with the same backend. The following two lines are adding a filter if the current query is for a multi-provider backend.